task/git-clone/0.1/tests/run.yaml

```diff
@@ -0,0 +1,30 @@
+---
```

Comment:
Have you considering using a Pipeline to encapsulate the Task execution and the required verification? The release folks took this approach, example: https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/check-data-keys/tests/test-check-data-keys-fail-missing-cdn-key.yaml

Reply:
@lcarva could you please help with more context around where these tests are running what approach they are following... any doc/link ?

Reply:
Here we should verify that the task really cloned the repo as expected (e.g. contains the expected files, is really at revision `1.0.0`, ...). And this test should set the example for how to verify that the task works as expected.

That will probably require wrapping the Task in a Pipeline as Lui suggested

Reply:
makes sense now, this is another approach to validate the tasks (we were not aware of while investigating), the approach taken release-service-catalog repo has more flexibility on what to verify in the task results. With current approach taken in this PR, we are just validating task results are non-empty.  If we want to choose the approach taken in release-service-catalog repo, the current PR needs to be rewritten almost.

Reply:
As a contributor to the release-service-catalog, I just want to say that I like the approach they're using. Once I got the hang of writing `mocks.sh` for my tests there, it became easy to add new functional tests for my tasks including both positive and negative cases.

They have nice features, like the ability to assert that the task should fail hard and exit with a non-zero status code on certain inputs, or to assert that the task should succeed, but have certain properties in its output.

For local development / inner-loop, I can stand up kind locally, install upstream tekton, and then run `export TEST_ITEMS=path/to/the/task/Im/working/on` and then run the `.github/scripts/test_tekton_tasks.sh` from their repo in a loop while I hack on my task or test.

Reply:
@ralphbean thanks for the feedback, looks like the approach used in release-service-catalog is better, will move to it.

Reply:
took this approach in [this PR](https://github.com/konflux-ci/build-definitions/pull/1644)  and closed the current one